src/fondant/cli.py

```diff
@@ -262,88 +261,108 @@ def register_run(parent_parser):
         pipeline (see fondant compile --help for more info)
         OR a path to a spec file in which case it will compile the pipeline first and then run it.
 
-        The --local or --kubeflow flag specifies the mode in which the pipeline will be ran.
-        You can use the --extra-volumes flag to specify extra volumes to mount in the containers this can be used:
+        You can use different modes for fondant runners. Current existing modes are `local` and `kubeflow`.
+        You can run `fondant <mode> --help` to find out more about the specific arguments for each mode.
 
-        Example:
-        fondant run my_project.my_pipeline.py --local --extra-volumes $HOME/.aws/credentials:/root/.aws/credentials
-        fondant run ./my_compiled_kubeflow_pipeline.tgz --kubeflow
+        Examples of running component:
+        fondant run local --extra-volumes $HOME/.aws/credentials:/root/.aws/credentials my_project.my_pipeline.py
+        fondant run kubeflow ./my_compiled_kubeflow_pipeline.tgz
         """,
         ),
     )
+
+    runner_subparser = parser.add_subparsers()
+    # Define the "ref" argument once
     parser.add_argument(
         "ref",
         help="""Reference to the pipeline to run, can be a path to a spec file or
             a module containing the pipeline instance that will be compiled first (e.g. pipeline.py)
             """,
         action="store",
     )
-    # add a mutually exclusive group for the mode
-    mode_group = parser.add_mutually_exclusive_group(required=True)
-    mode_group.add_argument("--local", action="store_true")
-    mode_group.add_argument("--kubeflow", action="store_true")
 
-    parser.add_argument(
+    local_parser = runner_subparser.add_parser(name="local", help="Local runner")
+    kubeflow_parser = runner_subparser.add_parser(
+        name="kubeflow",
+        help="Kubeflow runner",
+    )
+
+    # Local runner parser
+    local_parser.add_argument(
         "--output-path",
         "-o",
-        help="Output directory",
-        default=None,
+        help="Output path of compiled pipeline",
+        default="docker_compose.yml",
```

Comment:
See my comment above. 

```suggestion
        default="docker-compose.yml",
```